Failure to install on modern versions of Node.js (node-gyp compilation error from deprecated GetContents() function)

Related to the issue with pinning [nan `2.4.0`][1], this project fails to build on newer versions of Node.js.

The main compile error from `node-gyp` is related to `GetContents()` being deprecated in Node.js 14.

<details><summary>Expand for full compile log:</summary>
<p>

```
../qrusage.cc: In function ‘double* getFloat64ArrayPointer(unsigned int, v8::Local<v8::Value>)’:
../qrusage.cc:25:55: error: ‘class v8::ArrayBuffer’ has no member named ‘GetContents’
   25 |             double* fields = static_cast<double*>(ab->GetContents().Data());
      |                                                       ^~~~~~~~~~~
```

</p>
</details>

The fix can be seen in [nan `2.4.1`][2], which adds a C pre-processor directive checking for `(V8_MAJOR_VERSION >= 8)`.  If nodejs built-in V8 headers define this to be 8 or greater, then they use the new `GetBackingStore()` function.  Otherwise, they fall back on using `GetContents()`.

```c
// Actually it's 7.9 here but this would lead to ABI issues with Node.js 13
// using 7.8 till 13.2.0.
#if (V8_MAJOR_VERSION >= 8)
      data = static_cast<char*>(buffer->GetBackingStore()->Data()) + byte_offset;
#else
      data = static_cast<char*>(buffer->GetContents().Data()) + byte_offset;
#endif
```

The `V8_MAJOR_VERSION` definition can be found in `v8-version.h` from Node.js installation.  Usually this is in `/usr/include/node/v8-version.h`.



[1]: https://github.com/andrasq/node-qrusage/pull/1/commits/2f20703cbbc90b5c175be3618068e2efd40d9bae
[2]: https://github.com/nodejs/nan/blob/56585a1cd21f25b4d56168063cedaf0831c63fef/nan_typedarray_contents.h#L36-L40
[3]: https://github.com/nodejs/nan/pull/888

--- a/package.json
+++ b/package.json
@@ -16,7 +16,7 @@
   "keywords": [ "Andras", "quick", "fast", "getrusage", "cpu", "memory", "resources", "used" ],
 
   "dependencies": {
-    "nan": "2.14.0"
+    "nan": "^2.14.0"
   },
   "devDependencies": {
     "qnit": "^0.25"

--- a/qrusage.cc
+++ b/qrusage.cc
@@ -22,7 +22,13 @@ double * getFloat64ArrayPointer( unsigned int length, v8::Local<v8::Value> arg )
         v8::Local<v8::Float64Array> array = arg.As<v8::Float64Array>();
         if (array->Length() >= length) {
             v8::Local<v8::ArrayBuffer> ab = array->Buffer();
+// Actually it's 7.9 here but this would lead to ABI issues with Node.js 13
+// using 7.8 till 13.2.0.
+#if (V8_MAJOR_VERSION >= 8)
+            double* fields = static_cast<double*>(ab->GetBackingStore()->Data());
+#else
             double* fields = static_cast<double*>(ab->GetContents().Data());
+#endif
             return fields;
         }
     }

